name: "CI"
on:
  push:
    branches:
    - main
  pull_request:
    branches:
    - main
permissions:
  contents: read
  actions: write
concurrency:
  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
  cancel-in-progress: true
jobs:
  presubmit:
    name: "experimental_presubmit_test (py ${{ matrix.docker-python-version }} on linux-x86-n2-16)"
    runs-on: Linux_runner_8_core
    container:
      image: us-docker.pkg.dev/ml-oss-artifacts-published/ml-public-container/ml-build:latest
    if: |
      github.event.pull_request.user.type == 'Bot' ||
      github.event.pull_request.user.type == 'Organization' ||
      contains(github.event.pull_request.labels.*.name, 'external-pr-review')
    timeout-minutes: 60
    strategy:
      matrix:
        docker-python-version: ['3.11']
    steps:
      - name: Run bazel test
        # TODO(ecalubaquib): Remove continue-on-error once b/391391626 is fixed.
        continue-on-error: true
        run: |
          #!/bin/bash"
          cd ${GITHUB_WORKSPACE}
          cat .bazelrc | grep ml_build_config_platform
          export DOCKER_PYTHON_VERSION="${{ matrix.docker-python-version }}"
          export EXPERIMENTAL_TARGETS_ONLY="true"
          export BAZEL_CONFIG_FLAGS="--config=rbe_linux_cpu"
          export HERMETIC_PYTHON_VERSION=${DOCKER_PYTHON_VERSION}

          # TODO(ecalubaquib): Remove continue-on-error once b/391391626 is fixed.
          if [ $? -ne 0 ]; then
            echo "::warning:: Bazel test failed. Please check the logs for details."
          else
            echo "Bazel test passed."
          fi
